Userland/Demos/CatDog/CatDog.cpp

```diff
@@ -10,174 +10,187 @@
+        { State::RIGHT, "/res/icons/catdog/erun1.png"sv },
+        { State::IDLE | State::SLEEPING | State::FRAME_2, "/res/icons/catdog/sleep2.png"sv },
+        { State::IDLE | State::SLEEPING, "/res/icons/catdog/sleep1.png"sv },
+        { State::IDLE | State::ARTIST, "/res/icons/catdog/artist.png"sv },
+        { State::IDLE | State::INSPECTOR, "/res/icons/catdog/inspector.png"sv },
+        { State::IDLE, "/res/icons/catdog/still.png"sv },
+        { State::ALERT, "/res/icons/catdog/alert.png"sv }
+    };
+
     auto catdog = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) CatDog));
-    catdog->m_alert = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/alert.png"sv));
-    catdog->m_artist = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/artist.png"sv));
-    catdog->m_erun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/erun1.png"sv));
-    catdog->m_erun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/erun2.png"sv));
-    catdog->m_inspector = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/inspector.png"sv));
-    catdog->m_nerun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/nerun1.png"sv));
-    catdog->m_nerun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/nerun2.png"sv));
-    catdog->m_nrun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/nrun1.png"sv));
-    catdog->m_nrun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/nrun2.png"sv));
-    catdog->m_nwrun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/nwrun1.png"sv));
-    catdog->m_nwrun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/nwrun2.png"sv));
-    catdog->m_serun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/serun1.png"sv));
-    catdog->m_serun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/serun2.png"sv));
-    catdog->m_sleep1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/sleep1.png"sv));
-    catdog->m_sleep2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/sleep2.png"sv));
-    catdog->m_srun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/srun1.png"sv));
-    catdog->m_srun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/srun2.png"sv));
-    catdog->m_still = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/still.png"sv));
-    catdog->m_swrun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/swrun1.png"sv));
-    catdog->m_swrun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/swrun2.png"sv));
-    catdog->m_wrun1 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/wrun1.png"sv));
-    catdog->m_wrun2 = *TRY(Gfx::Bitmap::try_load_from_file("/res/icons/catdog/wrun2.png"sv));
-    catdog->m_curr_bmp = catdog->m_alert;
+    for (auto const& image_source : image_sources)
+        catdog->m_images.append({ image_source.state, *TRY(Gfx::Bitmap::try_load_from_file(image_source.path)) });
```

Comment:
As someone noted on Andreas's already-merged PR, this should be done with `TRY(catdog->m_images->try_append(TRY(... )));`

Although, we could probably get away with a construct like
```c++
TRY(catdog->m_images->try_reserve(image_sources.size()));
for (...)
   MUST( try_append)
```
... maybe? I'm not sure if this will work. If it doesn't compile, we should add a concept of `Vector<T>::ensure_capacity_uninitalized` or `ensure_capacity_for_overwrite` like C++20 did for std's vector at a later date.

Reply:
Fixed.